src/screens/browse/components/Modals/SourceSettings.tsx

```diff
@@ -112,20 +116,30 @@ const SourceSettingsModal: React.FC<SourceSettingsModal> = ({
       </Text>
       <Text style={[{ color: theme.onSurfaceVariant }]}>{description}</Text>
 
-      {Object.entries(pluginSettings).map(([key, setting]) => (
-        <TextInput
-          key={key}
-          mode="outlined"
-          label={setting.label}
-          value={formValues[key] || ''}
-          onChangeText={value => handleChange(key, value)}
-          placeholder={`Enter ${setting.label}`}
-          placeholderTextColor={theme.onSurfaceDisabled}
-          underlineColor={theme.outline}
-          style={[{ color: theme.onSurface }, styles.textInput]}
-          theme={{ colors: { ...theme } }}
-        />
-      ))}
+      {Object.entries(pluginSettings).map(([key, setting]) =>
+        setting?.type === 'Switch' ? (
+          <SwitchItem
+            key={key}
+            label={setting.label}
+            value={!!formValues[key]}
+            onPress={() => handleChange(key, formValues[key] ? '' : 'true')}
+            theme={theme}
+          />
+        ) : (
```

Comment:
switch setting type

(saves the values as either an empty string or the string "true" as plugin settings are all strings, and by making false falsy it will work just as well)